site/docs/concepts/architecture.md

```diff
@@ -2,27 +2,54 @@
-and whether restrictive actions such as masking or hashing have to be applied. Taking into account these governance actions, as well as application requirements, dataset specification, available infrastructure (5) and the [IT config policies](./config-policies.md) defined (4), the controller compiles a plotter (9). The plotter specifies a data plane connecting the application to the datasets it requires. More specifically, the plotter lists the modules to use, the capabilities required from these modules, the cluster on which each module should be deployed, as well as the flow of data between the asset and the workload through the chosen modules. For the plotter to be optimal in terms of the defined optimization goals (a.k.a. IT config soft policies), the controller may use a [CSP-based optimizer](./optimizer.md) (7).  If no CSP engine is installed, optimization goals will not be taken into account, and the manager will use the first (but not necessarily optimal) solution that meets all of the other requirements.
+These modules may be existing open source or third party services, or custom ones. The module of a service indicates the capabilities it supports, the formats and interfaces, and how to deploy the service.  Modules may describe externally deployed services, or services deployed by fybrik.  Examples of modules are those that provide read/write access or produce implicit copies that serve as lower latency caches of remote assets.  
+Modules may also perform actions to enforce data governance policy decisions, such as masking or redaction as examples.
 
-As data assets may reside in different clusters/clouds a `Blueprint` CRD is created for each cluster, containing the information regarding the services to be deployed or configured in the given cluster. Depending on the setup the `PlotterController` will use various methods to distribute the blueprints. In a multi cluster setup the default distribution implementation is using [Razee](http://razee.io) to control remote blueprints, but several multi-cloud tools
-could be used as a replacement. The `PlotterController` also collects statuses and distributes
-updates of said blueprints. Once all the blueprints on all clusters are ready the plotter is marked as ready, and the overall status is propagated back to the user in the `FybrikApplication` status.
+## [Connectors](./connectors.md)
+Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application.
+
+The current abstraction supports 4 different [connectors](./connectors.md): one for data catalog (2), one for policy manager (3), one for credential management (6) and one for CSP (7). It is designed in an open way so that multiple different catalog, policy, credentials and CSP frameworks of all kinds of cloud and on-prem systems can be supported.
```

Comment:
Please stay with 3 connectors and leave CSP out of it. 
There is no external connector for CSP.